File: georocket-cli/src/test/java/io/georocket/util/SizeFormatTest.java
package io.georocket.util;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * Test {@link SizeFormat}
 * @author Michel Kraemer
 */
public class SizeFormatTest {
  /**
   * Test if data sizes can be converted to human-readable strings
   */
  @Test
  public void format() {
    assertEquals("0 B", SizeFormat.format(-1));
    assertEquals("0 B", SizeFormat.format(0));
    assertEquals("100 B", SizeFormat.format(100));
    assertEquals("1 kB", SizeFormat.format(1000));
    assertEquals("1 kB", SizeFormat.format(1024));
    assertEquals("1.1 kB", SizeFormat.format(1000 + 100));
    assertEquals("1.2 kB", SizeFormat.format(1000 + 200));
    assertEquals("1.3 kB", SizeFormat.format(1000 + 260));
    assertEquals("1 MB", SizeFormat.format(1000 * 1000));
    assertEquals("2.5 MB", SizeFormat.format(1000 * 1000 * 2 + 1000 * 500));
    assertEquals("1 GB", SizeFormat.format(1000 * 1000 * 1000));
    assertEquals("1 TB", SizeFormat.format(1000L * 1000 * 1000 * 1000));
    assertEquals("1 PB", SizeFormat.format(1000L * 1000 * 1000 * 1000 * 1000));
    assertEquals("1 EB", SizeFormat.format(1000L * 1000 * 1000 * 1000 * 1000 * 1000));
    assertEquals("9.2 EB", SizeFormat.format(Long.MAX_VALUE));
  }
}

File: kotlin_impl/src/test/kotlin/io/georocket/util/SizeFormatTest.kt
package io.georocket.util

import io.georocket.util.SizeFormat.format
import org.junit.Assert
import org.junit.Test

/**
 * Test [SizeFormat]
 * @author Michel Kraemer
 */
class SizeFormatTest {
  /**
   * Test if data sizes can be converted to human-readable strings
   */
  @Test
  fun format() {
    Assert.assertEquals("0 B", format(-1))
    Assert.assertEquals("0 B", format(0))
    Assert.assertEquals("100 B", format(100))
    Assert.assertEquals("1 kB", format(1000))
    Assert.assertEquals("1 kB", format(1024))
    Assert.assertEquals("1.1 kB", format((1000 + 100).toLong()))
    Assert.assertEquals("1.2 kB", format((1000 + 200).toLong()))
    Assert.assertEquals("1.3 kB", format((1000 + 260).toLong()))
    Assert.assertEquals("1 MB", format((1000 * 1000).toLong()))
    Assert.assertEquals("2.5 MB", format((1000 * 1000 * 2 + 1000 * 500).toLong()))
    Assert.assertEquals("1 GB", format((1000 * 1000 * 1000).toLong()))
    Assert.assertEquals("1 TB", format(1000L * 1000 * 1000 * 1000))
    Assert.assertEquals("1 PB", format(1000L * 1000 * 1000 * 1000 * 1000))
    Assert.assertEquals("1 EB", format(1000L * 1000 * 1000 * 1000 * 1000 * 1000))
    Assert.assertEquals("9.2 EB", format(Long.MAX_VALUE))
  }
}
